generator/generator_tests/metadata_parser_test.cpp

```diff
@@ -0,0 +1,199 @@
+  TEST(md.Empty(), ());
+  p("stars", "8");
+  TEST(md.Empty(), ());
+  p("stars", "10");
+  TEST(md.Empty(), ());
+  p("stars", "910");
+  TEST(md.Empty(), ());
+  p("stars", "100");
+  TEST(md.Empty(), ());
+
+  // check correct values
+  p("stars", "1");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "1", ())
+  md.Drop(Metadata::FMD_STARS);
+
+  p("stars", "2");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "2", ())
+  md.Drop(Metadata::FMD_STARS);
+
+  p("stars", "3");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "3", ())
+  md.Drop(Metadata::FMD_STARS);
+
+  p("stars", "4");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "4", ())
+  md.Drop(Metadata::FMD_STARS);
+
+  p("stars", "5");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "5", ())
+  md.Drop(Metadata::FMD_STARS);
+
+  p("stars", "6");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "6", ())
+  md.Drop(Metadata::FMD_STARS);
+
+  p("stars", "7");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "7", ())
+  md.Drop(Metadata::FMD_STARS);
+
+  // check almost correct values
+  p("stars", "4+");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "4", ())
+  md.Drop(Metadata::FMD_STARS);
+
+  p("stars", "5s");
+  TEST_EQUAL(md.Get(Metadata::FMD_STARS), "5", ())
+  md.Drop(Metadata::FMD_STARS);
+
+}
+
+UNIT_TEST(Metadata_ValidateAndFormat_operator)
+{
+  classificator::Load();
+  Classificator const & c = classif();
+  uint32_t const type_atm = c.GetTypeByPath({ "amenity", "atm" });
+  uint32_t const type_fuel = c.GetTypeByPath({ "amenity", "fuel" });
+
+  FeatureParams params;
+  MetadataTagProcessor p(params);
+  Metadata & md = params.GetMetadata();
+
+  // ignore tag 'operator' if feature have inappropriate type
+  p("operator", "Some");
+  TEST(md.Empty(), ());
+
+  params.SetType(type_atm);
+  p("operator", "Some");
+  TEST_EQUAL(md.Get(Metadata::FMD_OPERATOR), "Some", ());
+  md.Drop(Metadata::FMD_OPERATOR);
+
+  params.SetType(type_fuel);
+  p("operator", "Some");
+  TEST_EQUAL(md.Get(Metadata::FMD_OPERATOR), "Some", ());
+  md.Drop(Metadata::FMD_OPERATOR);
+
+  params.SetType(type_atm);
+  params.AddType(type_fuel);
+  p("operator", "Some");
+  TEST_EQUAL(md.Get(Metadata::FMD_OPERATOR), "Some", ());
+  md.Drop(Metadata::FMD_OPERATOR);
+}
+
+UNIT_TEST(Metadata_ValidateAndFormat_ele)
+{
+  classificator::Load();
+  Classificator const & c = classif();
+  uint32_t const type_peak = c.GetTypeByPath({ "natural", "peak" });
+
+  FeatureParams params;
+  MetadataTagProcessor p(params);
+  Metadata & md = params.GetMetadata();
+
+  // ignore tag 'operator' if feature have inappropriate type
+  p("ele", "123");
+  TEST(md.Empty(), ());
+
+  params.SetType(type_peak);
+  p("ele", "0");
+  TEST(md.Empty(), ());
+
+  params.SetType(type_peak);
+  p("ele", "0,0000");
+  TEST(md.Empty(), ());
+
+  params.SetType(type_peak);
+  p("ele", "0.0");
+  TEST(md.Empty(), ());
+
+  params.SetType(type_peak);
+  p("ele", "123");
+  TEST_EQUAL(md.Get(Metadata::FMD_ELE), "123", ());
+  md.Drop(Metadata::FMD_ELE);
+}
+
+#ifdef OMIM_OS_MOBILE
```

Comment:
As WIKIHOST is used inside Metadata_ValidateAndFormat_wikipedia, please, move these defines inside this test. And add 
```
#undef WIKIHOST
```
At the end of the test.